src/lib/utils/os_utils.cpp

```diff
@@ -163,6 +167,36 @@ uint64_t OS::get_cpu_cycle_counter()
    return rtc;
    }
 
+size_t OS::get_cpu_total()
+   {
+   size_t tt = 0;
+#if defined(BOTAN_TARGET_OS_HAS_POSIX1) && defined(_SC_NPROCESSORS_CONF)
+   long res;
+   if ((res = ::sysconf(_SC_NPROCESSORS_CONF)) <= 0)
+     {
+          return 1;
+     }
+     tt = static_cast<size_t>(res);
+#else
```

Comment:
This should be guarded with `HAS_THREADS`, then a final `#else` (for both no threads and no sysconf) that just sets tt to 1.